pkg/virt-handler/node-labeller/cpu_plugin.go

```diff
@@ -0,0 +1,196 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package nodelabeller
+
+import (
+	"encoding/xml"
+	"io/ioutil"
+	"os/exec"
+	"strings"
+)
+
+const (
+	usableNo      string = "no"
+	DefaultMinCPU        = "Penryn"
+)
+
+var DefaultObsoleteCPUs = map[string]bool{
+	"486":        true,
+	"pentium":    true,
+	"pentium2":   true,
+	"pentium3":   true,
+	"pentiumpro": true,
+	"coreduo":    true,
+	"n270":       true,
+	"core2duo":   true,
+	"Conroe":     true,
+	"athlon":     true,
+	"phenom":     true,
+}
+
+// getCPUInfo processes all cpu info data and returns
+// slice of usable cpu models and features.
+func (n *NodeLabeller) getCPUInfo() ([]string, map[string]bool) {
+	minCPU := n.clusterConfig.GetMinCPUModel()
+	if minCPU == "" {
+		minCPU = DefaultMinCPU
+	}
+
+	obsoleteCPUsx86 := n.clusterConfig.GetObsoleteCPUModels()
+	if obsoleteCPUsx86 == nil {
+		obsoleteCPUsx86 = DefaultObsoleteCPUs
+	}
+
+	basicFeaturesMap := n.cpuInfo.models[minCPU]
+
+	cpus := make([]string, 0)
+	features := make(map[string]bool)
+	newFeatures := make(map[string]bool)
+
+	for model := range n.hostCapabilities.items {
+		if val, ok := obsoleteCPUsx86[model]; ok && val {
+			continue
+		}
+
+		newFeatures = subtractMap(n.cpuInfo.models[model], basicFeaturesMap)
+		features = unionMap(features, newFeatures)
```

Comment:
The `unionMap()` function creates a new map on each iteration. This code can be more efficient if each iteration only adds elements to the existing `features` map.